packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -54,13 +61,47 @@ function AuthOptions({
     hasLoggedOut() ? Display.SignBack : defaultDisplay,
   );
 
-  const onProviderClick = (provider: string) => {
-    onSelectedProvider({
+  const { data: registration } = useQuery(
+    'registration',
+    initializeRegistration,
+    { ...disabledRefetch },
+  );
+
+  useEffect(() => {
+    window.onmessage = async function receiveMessage(e) {
```

Comment:
Let's probably create a hook adding event listeners on window-related handlers. It might involve the same parameters for the `addEventListener` argument. So we can just add handlers without even thinking about the cleanups since it will be managed by this hook.

Reply:
Ah good point 🤘
I'll work on that

Reply:
@sshanzel Added, let me know what you think